metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -0,0 +1,1152 @@
+        foreachSeq(_.autoConnectToBuildServer())
+      case ServerCommands.GenerateBspConfig() =>
+        foreachSeq(
+          _.generateBspConfig(),
+          ignoreValue = true,
+        )
+      case ServerCommands.ImportBuild() =>
+        foreachSeq(
+          _.slowConnectToBuildServer(forceImport = true)
+        )
+      case ServerCommands.ConnectBuildServer() =>
+        foreachSeq(_.quickConnectToBuildServer())
+      case ServerCommands.DisconnectBuildServer() =>
+        foreachSeq(_.disconnectOldBuildServer(), ignoreValue = true)
+      case ServerCommands.DecodeFile(uri) =>
+        getServiceFor(uri).decodeFile(uri).asJavaObject
+      case ServerCommands.DiscoverTestSuites(params) =>
+        getServiceFor(params.uri).discoverTestSuites(params.uri).asJavaObject
+      case ServerCommands.DiscoverMainClasses(unresolvedParams) =>
+        getServiceFor(unresolvedParams.path)
+          .discoverMainClasses(unresolvedParams)
+          .asJavaObject
+      case ServerCommands.RunScalafix(params) =>
+        val uri = params.getTextDocument().getUri()
+        getServiceFor(uri).runScalafix(uri).asJavaObject
+      case ServerCommands.ChooseClass(params) =>
+        val uri = params.textDocument.getUri()
+        val searchGranularity =
+          if (params.kind == "class") ClassFinderGranularity.ClassFiles
+          else ClassFinderGranularity.Tasty
+        getServiceFor(uri).chooseClass(uri, searchGranularity).asJavaObject
+      case ServerCommands.RunDoctor() =>
+        onCurrentFolder(_.rundoctor(), "run doctor").asJavaObject
+      case ServerCommands.ZipReports() =>
+        Future {
+          val zip =
+            ZipReportsProvider.zip(folderServices.map(_.folderReportsZippper))
+          val pos = new lsp4j.Position(0, 0)
+          val location = new Location(
+            zip.toURI.toString(),
+            new lsp4j.Range(pos, pos),
+          )
+          languageClient.metalsExecuteClientCommand(
+            ClientCommands.GotoLocation.toExecuteCommandParams(
+              ClientCommands.WindowLocation(
+                location.getUri(),
+                location.getRange(),
+              )
+            )
+          )
+        }.asJavaObject
+      case ServerCommands.ListBuildTargets() =>
+        Future {
+          folderServices
+            .flatMap(
+              _.buildTargets.all.toList
+                .map(_.getDisplayName())
+                .sorted
+            )
+            .asJava
+        }.asJavaObject
+      case ServerCommands.BspSwitch() =>
+        foreachSeq(_.switchBspServer())
+      case ServerCommands.OpenIssue() =>
+        Future
+          .successful(Urls.openBrowser(githubNewIssueUrlCreator.buildUrl()))
+          .asJavaObject
+      case OpenBrowserCommand(url) =>
+        Future.successful(Urls.openBrowser(url)).asJavaObject
+      case ServerCommands.CascadeCompile() =>
+        foreachSeq(_.cascadeCompile(), ignoreValue = true)
```

Comment:
Both clean and casade compiler should happen for the current folder I think. Otherwise it might be too intensive.